src/test/kotlin/org/rust/ide/hints/RsChainMethodTypeHintsProviderTest.kt

```diff
@@ -112,6 +112,19 @@ class RsChainMethodTypeHintsProviderTest : RsInlayTypeHintsTestBase() {
         }
     """)
 
+    @ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
+    fun `test iterator special case`() = doTest("""
+        fn main() {
+            vec![1, 2, 3]
+                .into_iter()/*hint text="[:  [impl  [Iterator [< [Item = i32] >]] ]]"*/
+                .map(|x/*hint text="[:  i32]"*/| x as u8)/*hint text="[:  [impl  [Iterator [< [Item = u8] >]] ]]"*/
```

Comment:
Hmm, is there a way to disable regular type hints in these tests?

Reply:
Looks like there isn't API to do it
I wanted to make this improvement but haven't tried yet

Reply:
https://github.com/Kobzol/intellij-rust/commit/5af251752cce6c7e3b220f38f7db07730f0705f1 this seems to work. BTW the same situation is in inlay type hints, where I had to change one test to add chain method hints, so if this works it could also be applied there.

Reply:
@vlad20012 @Undin Probably you should use here provider specific test case: `com.intellij.testFramework.utils.inlays.InlayHintsProviderTestCase`, it allows to test different settings and only a given provider